### PR TITLE
Feature #157

### PR DESCRIPTION
**Pull request by wasikuss**
Changes to achieve effect described in #157.

capture

### CODE REVIEW

File: lib/remote-ftp.js [hunk 1]
```diff
@@ -400,9 +404,13 @@ module.exports = {
 
 		var self = this;
 
-		//if (!atom.project.remoteftp.isConnected())
-		//	return;
-
+		if (!atom.project.remoteftp.isConnected()) {
+			if(atom.config.get('Remote-FTP.autoConnectOnSave'))
```
Comment (jpxd) [inline lib/remote-ftp.js:408]: First of all I don't think it's good style to encapsulate another `if` here. I would just use:

```
if (!atom.project.remoteftp.isConnected() && !atom.config.get('Remote-FTP.autoConnectOnSave'))
    return;
```

You don't need to connect here, `atom.project.remoteftp.upload(...)` does this for you.
You can just do a code cleanup commit on this or I just merge the few lines manually :)

**Comment by jpxd** [comment]
Merged manually in 20f7242eea53bbabff25f2b1cc2e56871df1ba2c

**Comment by wasikuss** [comment]
After you changes it also won't work. If I save file `remote-ftp` should connnect to ftp server ( `Auto Connect` - ON, `Auto Upload` - OFF )

```
...
fileSaved: function (text) {
    if (!hasProject()) return;

var self = this;

if (!atom.project.remoteftp.isConnected()) {
    if(atom.config.get('Remote-FTP.autoConnectOnSave'))
        atom.project.remoteftp.connect();
    else
        return;
}

if (!atom.config.get('Remote-FTP.autoUploadOnSave')) return;

var local = text.path;
...
```

Code above passes 4 cases correctly
1) ( `Auto Upload` - OFF, `Auto Connect` - OFF ): do nothing on save
1) ( `Auto Upload` - OFF, `Auto Connect` - ON ): on save open connection to allow manual upload
1) ( `Auto Upload` - ON, `Auto Connect` - OFF ): on save upload if connected, if not - do nothing
1) ( `Auto Upload` - ON, `Auto Connect` - ON ): on save upload, and if disconnected then make new connection

**Comment by jpxd** [comment]
But why would you want to auto-connect on file save and don't upload
anything ?
If you manually upload a file, the connection gets established
automatically anyway.

This setting was just, to allow better control when a file has to be
uploaded automatically on save (Always, Only when already connected, never)

**Comment by jpxd** [comment]
What do you think of using a dropdown for this @wasikuss ?

screenshot 2015-08-17 10 34 49

Options: always, only when connected, never

**Comment by wasikuss** [comment]
Yes, it is easier to understand than analyzing states of two options. I'm in favor to make the changes and already using this solution.

**Comment by jpxd** [comment]
Done in d334325f02e208ccbe039f6106b3327e384e6d6a

**Pull request closed without merging**